### PR TITLE
[Metal] Pass random seed to metal backend

### PR DESCRIPTION
**Pull request by knight42**
Signed-off-by: knight42 <anonymousknight96@gmail.com>

Related issue = #2734

/cc @k-ye 

I am not sure if the current implementation is desired since I don't know much about cpp 😢 , any advice is welcome!

### CODE REVIEW

File: tests/python/test_random.py [hunk 1]
```diff
@@ -2,11 +2,7 @@
 from taichi import approx
 
 
-def archs_support_random(func):
```
Comment (knight42) [inline tests/python/test_random.py:5]: I have grep-ed this function in the project and it seems this function is not used anywhere, so I delete it in this PR. Please feel free to correct me if I am missing something.

File: tests/python/test_random.py [hunk 2]
```diff
@@ -128,7 +124,7 @@ def gen():
     assert not np.allclose(result[0], result[1])
 
 
-@ti.test(arch=[ti.cpu, ti.cuda])
+@ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
```
Comment (k-ye) [inline tests/python/test_random.py:127]: Our CI's Metal doesn't support 64-bit data yet

```suggestion
@ti.test(arch=[ti.cpu, ti.cuda])
```

Comment (knight42) [reply tests/python/test_random.py]: Got it!

**Comment by knight42** [comment]
@k-ye Hi! All tests passed now! Would you like to take a second look?

**Review by k-ye — APPROVED** [review]
LGTM Thanks!

**Pull request merged**